### DISCUSSION TITLE
Decide on UI for DxStation's CW Speed adjustment

### DISCUSSION THREAD

**Discussion by w7sst** [w7sst/MorseRunner#84] [Ideas]
It turns out that N2IC, F6FVY, and most recently JM1DXH have all made changes to adjust the CW Speed of the DxStation. MorseRunner original UI was changing setting the sending speed and then applying a random offset to that speed. However, sometimes this speed is slow for a user who is training and trying to improve their copying speed.

This discussion will examine the three implementations and choose one.

Briefly...
1) N2IC - not sure of how Steve solved this. I'll have to check and update this description.
2) F6FVY - David added two menu items to Settings menu, for setting a positive and negative adjustment to the sending CW speed. The DxStation will then send using a random value within this range.
3) JM1DXH - Add a new spin control to the UI for adjusting the range of DxStation CW Sending speed. I have asked for a picture of this UI change and will add it here once I have it.

Our discussion should focus on useability of these choices. Keep in mind that someday I think we will explore a dedicated Settings Dialog. This will allow most Settings menu items and UI controls to be moved off of the main UI into a dedicated Settings UI.

Please comment below on the merits of the 3 choices above.

**Comment by w7sst** [comment]
JM1DXH added a description of their CW Speed adjustment solution [here](https://github.com/jm1dxh/MorseRunner/wiki/MorseRunner-Version-1.68.3). It uses two spin controls (one each for min and maximum CW speed):
![image](https://user-images.githubusercontent.com/106113737/198895033-1a34c385-92bb-4af6-9617-ba30201e56b8.png)

In comparison, David, @ct7aup, added two menu items:
![image](https://user-images.githubusercontent.com/106113737/198895169-335bc46a-98a0-47ac-bbec-f955ddf3a3e4.png)

Both accomplish the same goal of setting a min and max value for the Dx Station CW sending speed.

To summarize, there are several choices for setting the min/max CW Dx Station CW sending speed:
- use menu items (as shown above)
- add min and max spin controls (as shown above)
- adding a settings dialog (which has more area for setting these values)
- use a slider control with two values in the same control (min and max)
- use a menu item, `Settings | CW Sending Speed...` to bring up a dialog and ask user for min and max values.

Thoughts?

**Comment by ct7aup** [comment]
Hi Mike,

My goal in using relative values instead of absolute values was to allow changing the send speed without having to change the minimum and maximum receive speed values.

This could be also made on the JM1DXH interface by mantaining the same displacement of the max and minimum speed everytime the change of the sending speed happens. This  could be a very flexible way to set the speeds.

But in a user interface perspective your proposal of using a slider control seems the best solution.

**Comment by scotthibbs** [comment]
After watching Jay's W4ZXT's video found here: https://www.youtube.com/watch?v=DaWcZjucmuo

He seems to want this control:
![image](https://user-images.githubusercontent.com/13142469/200927826-266f47f7-611e-4905-b5c0-b0e658ffbc29.png)

I Agree.
